### ISSUE TITLE
Pagefind breaks in production

### ISSUE THREAD

**Issue by jonnysmillie** [withastro/starlight#1157]
### What version of `starlight` are you using?

0.11.1

### What version of `astro` are you using?

3.2.3

### What package manager are you using?

npm

### What operating system are you using?

N/A

### What browser are you using?

N/A

### Describe the Bug

When deploying a Starlight site to production the Pagefind search seems to break and I get these errors:

![Screenshot 2023-11-27 at 21 31 28](https://github.com/withastro/starlight/assets/7614348/bf0e4a6f-52f7-4d18-8bce-920dc8c6d48a)

The search works fine locally when I build the site and preview. It looks like there is some issue with Pagefind on the server?

I tried this with a new Starlight project to check and I got the same result.

### Link to Minimal Reproducible Example

https://defiant-invention.surge.sh/

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for reporting! It looks like in production Surge is responding with 410 Errors for some of Pagefind’s metadata files. I’m not really sure what could be causing that. Might be an issue with Surge itself?

**Comment by jonnysmillie** [comment]
Thank you for your response. Yes I think it could be something with Surge but I am also experiencing the issue with Netlify. I think it is more of a Pagefind issue related to this: https://github.com/CloudCannon/pagefind/issues/460.

I will update if I find a solution.

**Comment by delucis** [comment]
Oh interesting! We were deploying Starlight’s docs to Netlify until recently, so I know it _can_ work there — not sure why it isn’t in this case 🤔 

Do you see any logging during the build related to Pagefind?

**Comment by jonnysmillie** [comment]
Yes! Here you are, it seems that Netlify skips Pagefind due to linux-x64. I think this is unrelated (but similar) to the  Surge issue.

![Screenshot 2023-11-28 at 09 23 53](https://github.com/withastro/starlight/assets/7614348/16f60ef2-781f-432c-b492-6d8f6fcebc1e)

**Comment by delucis** [comment]
Thanks for sharing. I wonder if it’s just a flaky build failing on that download for some reason 🤔 

I might try triggering a build with a clean cache to see if that helps. In Netlify you can do that from the Deploys dashboard:

<img width="284" alt="Trigger Deploy dropdown showing Clear cache and deploy site option" src="https://github.com/withastro/starlight/assets/357379/b79b0044-d8c9-4a4f-899d-e402da62b47e">
